src/EditorFeatures/Core/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -20,6 +20,13 @@ namespace Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.AsyncComplet
 {
     internal sealed partial class ItemManager : IAsyncCompletionItemManager2
     {
+        /// <summary>
+        /// The threshold for us to consider including (potentially large amount of) expanded items in completion list.
+        /// Showing a large amount of expanded items to user would introduce noise and render the list too long to browse. 
+        /// Not processing those expanded items also has perf benefit (e.g. matching and highlighting could be expensive)
+        /// </summary>
+        internal const int MaximumFilterTextLengthToExcludeExpandedItems = 3;
```

Comment:
can we *not* do this if the filter text is all upper case.  if i do ST, or HS i do want things like SyntaxToken and HashSet.  Maybe all caps is a strong signal that this is camel-case-matching and we can include the rest?

Reply:
Adjust the threadhold to FilterText.Length < 2